ipn/ipnserver/server.go

```diff
@@ -120,6 +122,10 @@ func (s *Server) awaitBackend(ctx context.Context) (_ *ipnlocal.LocalBackend, ok
 // This is primarily for the Windows GUI, because wintun can take awhile to
 // come up. See https://github.com/tailscale/tailscale/issues/6522.
 func (s *Server) serveServerStatus(w http.ResponseWriter, r *http.Request) {
+	if !buildfeatures.HasDebug && runtime.GOOS != "windows" {
```

Comment:
Just for my understanding, we are checking GOOS here just so that Windows callers don't get a 404 on this endpoint even when the debugger is nominally disabled, I think.  If so, what's the advantage of reporting an error for other platforms? If the left arm of the condition is true (i.e., debug disabled) doesn't this thwart the linker's ability to remove the code below because the right arm is not constant?

(Or is `runtime.GOOS` actually constant too under the covers?)